model/task_lifecycle.go

```diff
@@ -434,7 +434,8 @@ func MarkEnd(t *task.Task, caller string, finishTime time.Time, detail *apimodel
 		grip.Warningf("Tried to mark task %s as finished twice", t.Id)
 		return nil
 	}
-	if detailsCopy.Status == evergreen.TaskSucceeded && t.MustHaveResults {
+	if detailsCopy.Status == evergreen.TaskSucceeded && t.MustHaveResults && !t.HasResults {
```

Comment:
It seems like the `MustHaveResults` check won't work anymore once test results are sent to cedar unless we have a way to ask Cedar whether or not the test results are present. Thoughts on this @john-m-liu?

Reply:
This gets sent on a successful send to cedar in the agent code.

Reply:
Hmm okay I re-read the code and it seems like we're relying on specifically [this check](https://github.com/evergreen-ci/evergreen/blob/a7c8faf0676c3fff1ffbf93a62fc93f4ab407f5a/agent/command/results_utils.go#L22-L24) to fail fast before we can send empty results to Cedar. Can we also make it so that Cedar errors if the server receives empty test results?